src/levanter/models/llama.py

```diff
@@ -0,0 +1,521 @@
+from dataclasses import dataclass
+from typing import Callable, Dict, Optional, Tuple, Union
+
+import equinox as eqx
+import jax
+import jax.numpy as jnp
+import jax.random as jrandom
+from transformers import LlamaConfig as HfLlamaConfig
+from transformers import PretrainedConfig as HfConfig
+
+import haliax as hax
+import haliax.nn as hnn
+from haliax import Axis, AxisSpec, NamedArray
+from haliax.jax_utils import named_call, shaped_rng_split
+from haliax.nn.scan import Stacked
+
+from levanter.compat.hf_checkpoints import HFCheckpointConverter
+from levanter.compat.torch_serialization import (
+    StateDict,
+    StateDictSerializationMixin,
+    apply_prefix,
+    flatten_linear_layers,
+    stack_state_dict,
+    unflatten_linear_layers,
+    unstack_state_dict,
+)
+from levanter.models.gpt2 import ACT2FN
+from levanter.models.lm_model import LmConfig
+from levanter.utils.py_utils import cached_classproperty
+
+
+@LmConfig.register_subclass("llama")
+@dataclass(frozen=True)
+class LlamaConfig:
+    """Config for LlamaModel
+
+    Args:
+        seq_len (int, optional): maximum length of the input sequence. Defaults to 2048.
+        hidden_dim (int, optional): dimension of the hidden state. Defaults to 4096.
+        intermediate_dim (int, optional): dimension of the intermediate state. Defaults to 11008.
+        num_layers (int, optional): number of hidden layers in the Transformer encoder. Defaults to 32.
+        num_heads (int, optional): number of attention heads for each attention layer. Defaults to 32.
+        activation_function (str, optional): activation function for the hidden layer. Defaults to "silu".
+        rope_scaling (Dict, optional): dict containing the scaling configuration for the Rotary Positional Embedding.
+    """
+
+    # TODO(ivan): add tying of embeddings, default it to false to match the original model
```

Comment:
@Ivan-Zhou this is critical. We probably need to add a roundtrip test starting from an HF model to catch this.